src/test/scala/fi/oph/scalaschema/ValidationAndExtractionTest.scala

```diff
@@ -326,6 +328,128 @@ class ValidationAndExtractionTest extends AnyFreeSpec with Matchers {
+          verifyValidation[NotWhenMixedListFieldContainer](JObject("number" -> JInt(4), "thing" -> JObject("field" -> JString("bogus"))), Right(NotWhenMixedListFieldContainer(4, WithNotWhenMixedListFieldsInParent(Some("bogus")))))
+        }
+
+        "Allows field only when value does not match the single value" in {
+          val expectedError = NotWhenMismatch(List(SerializableNotWhen("../number", JInt(1))))
+          verifyValidation[NotWhenSingleElementFieldContainer](JObject("number" -> JInt(1), "thing" -> JObject("field" -> JString("bogus"))), Left(List(ValidationError("thing.field", JString("bogus"), expectedError))))
+          verifyValidation[NotWhenSingleElementFieldContainer](JObject("number" -> JInt(2), "thing" -> JObject("field" -> JString("bogus"))), Right(NotWhenSingleElementFieldContainer(2, WithNotWhenSingleElementFieldsInParent(Some("bogus")))))
+        }
+
+        "Works with @DefaultValue and multiple allowed alternatives" in {
+          val expectedError = NotWhenMismatch(List(SerializableNotWhen("code", JArray(List(JInt(1), JInt(2)))), SerializableNotWhen("code", JArray(List(JInt(3), JInt(4))))))
+          verifyValidation[WithNotWhenFieldsWithDefaultValueAndMultipleAllowedValues](JObject("code" -> JInt(1), "field" -> JString("Something1")), Left(List(ValidationError("field", JString("Something1"), expectedError))))
+          verifyValidation[WithNotWhenFieldsWithDefaultValueAndMultipleAllowedValues](JObject("code" -> JInt(2), "field" -> JString("Something2")), Left(List(ValidationError("field", JString("Something2"), expectedError))))
+          verifyValidation[WithNotWhenFieldsWithDefaultValueAndMultipleAllowedValues](JObject("code" -> JInt(3), "field" -> JString("Something3")), Left(List(ValidationError("field", JString("Something3"), expectedError))))
+          verifyValidation[WithNotWhenFieldsWithDefaultValueAndMultipleAllowedValues](JObject("code" -> JInt(4), "field" -> JString("Something4")), Left(List(ValidationError("field", JString("Something4"), expectedError))))
+          verifyValidation[WithNotWhenFieldsWithDefaultValueAndMultipleAllowedValues](JObject("code" -> JInt(5), "field" -> JString("Something")), Right(WithNotWhenFieldsWithDefaultValueAndMultipleAllowedValues(5, "Something")))
+        }
+
+        "Allows a value matching @DefaultValue even when @NotWhen says it's not ok" in {
+          verifyExtractionRoundTrip[WithNotWhenFieldsWithDefaultValueAndMultipleAllowedValues](WithNotWhenFieldsWithDefaultValueAndMultipleAllowedValues(5, "DefaultVal"))
+        }
+
+        "Allows null-checking by using None as value" in {
+          verifyValidation[FieldNotOkIfNotWhenParentMissing](JObject("field" -> JString("hello")), Left(List(ValidationError("field", JString("hello"), NotWhenMismatch(List(SerializableNotWhen("..", JNull)))))))
+          verifyValidation[WrapperForTraitOkIfNotWhenParentMissing](
```

Comment:
I don't understand why this second one fails and third one doesn't, let's discuss these f2f.